controllers/restaurant.go

```diff
@@ -50,3 +46,45 @@ func GetRestaurant(w http.ResponseWriter, r *http.Request) {
 	resp.Data = restaurant
 	w.Write(resp.JSONBytes())
 }
+
+func updateOneRestaurant(id string, body map[string]string, resp *utils.RespMsg, w http.ResponseWriter) {
+	err := models.UpdateOneRestaurant(id, body)
+	if err != nil {
+		fmt.Println(err)
+		resp.Code = http.StatusInternalServerError
+		resp.Message = "server error"
+		w.Write(resp.JSONBytes())
+		return
+	}
+	resp.Code = http.StatusOK
+	w.Write(resp.JSONBytes())
+	return
+}
+
+// Restaurant will handle get/update/delete one restaurant
+func Restaurant(w http.ResponseWriter, r *http.Request) {
```

Comment:
The name here isn't clear to me what it's doing. How about 
```suggestion
func ShowOrUpdate(w http.ResponseWriter, r *http.Request) {
```

Also I would put the exported functions at the top above unexported functions

Reply:
I believe the delete endpoint will go here in the future

Reply:
I would argue that we're violating YAGNI (https://martinfowler.com/bliki/Yagni.html) but fair enough 😁